pkg/util/azureclient/applens/applens_client_options.go

```diff
@@ -40,11 +55,8 @@ func NewClientOptions() *ClientOptions {
 			},
 			Transport: &http.Client{
 				Transport: &http.Transport{
-					TLSNextProto: make(map[string]func(authority string, c *tls.Conn) http.RoundTripper),
-					TLSClientConfig: &tls.Config{
-						Renegotiation: tls.RenegotiateFreelyAsClient,
-						MinVersion:    tls.VersionTLS12,
-					},
+					TLSNextProto:    make(map[string]func(authority string, c *tls.Conn) http.RoundTripper),
+					TLSClientConfig: tlsConfig,
```

Comment:
nit: since `rootCAs` is nil by default, we can skip the nil check on lines `26-38` and just define it here as
```suggestion
					TLSClientConfig: &tls.Config{
						RootCAs:       certPool,
						Renegotiation: tls.RenegotiateFreelyAsClient,
						MinVersion:    tls.VersionTLS12,
					},,
```